config/reader.go

```diff
@@ -561,6 +685,10 @@ func getFromCloudGRPC(ctx context.Context, cloudCfg *Cloud, logger golog.Logger)
 		return nil, shouldCheckCacheOnFailure, err
 	}
 
+	if err := replacePlaceholdersInCloudConfig(cfg); err != nil {
```

Comment:
does this when we actually pull down the cloud config so that we don't need to check for it all over the reader code.